core/src/wallet/cosmos/explorers/GaiaCosmosLikeBlockchainExplorer.cpp

```diff
@@ -274,5 +274,153 @@ namespace ledger {
+                {"Content-Type", "application/json"}};
+            const bool jsonParseNumbersAsString = true;
+
+            return _http->GET(endpoint, headers)
+                .json(jsonParseNumbersAsString)
+                .mapPtr<BigInt>(
+                    getContext(),
+                    [endpoint](const HttpRequest::JsonResult &result) -> std::shared_ptr<BigInt> {
+                        auto &json = *std::get<1>(result);
+                        // TODO : clean error handling
+
+                        const auto &del_val_entries = json.GetObject()["result"].GetArray();
+                        BigInt total_amt = BigInt::ZERO;
+                        for (const auto& delegation_entry : del_val_entries) {
+                            total_amt = total_amt + BigInt::fromDecimal(delegation_entry.GetObject()["balance"].GetString());
+                        }
+
+                        return std::make_shared<BigInt>(total_amt);
+                    });
+        }
+        /// Get total pending rewards
+        FuturePtr<BigInt> GaiaCosmosLikeBlockchainExplorer::getPendingRewards(
+            const std::string &account) const
+        {
+            const auto endpoint = fmt::format("/distribution/delegators/{}/rewards", account);
+
+            std::unordered_map<std::string, std::string> headers{
+                {"Content-Type", "application/json"}};
+            const bool jsonParseNumbersAsString = true;
+
+            return _http->GET(endpoint, headers)
+                .json(jsonParseNumbersAsString)
+                .mapPtr<BigInt>(
+                    getContext(),
+                    [endpoint](const HttpRequest::JsonResult &result) -> std::shared_ptr<BigInt> {
+                        auto &json = *std::get<1>(result);
+
+                        if (!json.IsObject() ||
+                            !json.GetObject().HasMember("result") ||
+                            !json.GetObject()["result"].IsObject() ||
+                            !json.GetObject()["result"].GetObject().HasMember("total") ||
+                            !json.GetObject()["result"].GetObject()["total"].IsArray()
+                            ) {
+                            throw make_exception(
+                                api::ErrorCode::HTTP_ERROR,
+                                fmt::format("Failed to get total for {}", endpoint));
+                        }
+
+                        if (json.GetObject()["result"].GetObject()["total"].GetArray().Size() <= 0) {
+                            return std::make_shared<BigInt>(BigInt::ZERO);
+                        }
+
+                        std::string floating_amount = json.GetObject()["result"].GetObject()["total"]
+                            .GetArray()[0]
+                            .GetObject()["amount"]
+                            .GetString();
+                        floating_amount.erase(floating_amount.find("."), std::string::npos);
```

Comment:
We may want to round instead of truncate if adding many amounts, but probably irrelevant for now.

Reply:
In any case we're getting the total here, so there's nothing to sum anymore.

This result is the only one which gives non integer amounts (the amounts here are in `uatom` everytime). I think it's safe to have a balance here that's wrong by 1uatom max.

Reply:
```json
{
  "height": "0",
  "result": {
    "rewards": [
      {
        "validator_address": "cosmosvaloper1sjllsnramtg3ewxqwwrwjxfgc4n4ef9u2lcnj0",
        "reward": [
          {
            "denom": "uatom",
            "amount": "0.197274882002377000"
          }
        ]
      },
      {
        "validator_address": "cosmosvaloper1clpqr4nrk4khgkxj78fcwwh6dl3uw4epsluffn",
        "reward": [
          {
            "denom": "uatom",
            "amount": "0.334526290478806600"
          }
        ]
      }
    ],
    "total": [
      {
        "denom": "uatom",
        "amount": "0.531801172481183600"
      }
    ]
  }
}
```

Reply:
My bad, no addition here indeed! It's fine then.